app/routes/jobs.$id._index.tsx

```diff
@@ -40,11 +43,41 @@ export const action = async ({ params, request }: ActionFunctionArgs) => {
 
 export default function JobPage() {
   const { job } = useLoaderData<typeof loader>();
+  const { revalidate } = useRevalidator();
+
+  useEffect(() => {
+    // set interval to refresh job status
+    const id = setInterval(() => {
+      // reload only if user is on the page/tab
+      if (
+        document.visibilityState === "visible" &&
+        // and job state is not "definitive"
+        job.state !== "error" &&
```

Comment:
I tested to see what happens when I keep window visible the whole time.  I did not expect that is would show completed job page after a while, but it did show the browse page. Probably the loader does the redirect so we never end up here when the job goes from running to ok.